src/classes/RD2_OpportunityEvaluationService.cls

```diff
@@ -265,22 +265,26 @@ public inherited sharing class RD2_OpportunityEvaluationService {
             if (rdRecord.isClosed()) {
                 closedRDs.add(rd);
 
-            } else if (isNewInstallmentCandidate(rdRecord)) {
-                newOppRDs.add(rd);
-
-            } else if (isUpdateContext(rdRecord)) {
-                updateOppRDs.add(rd);
+            } else {
+                if (isNewOpportunityCandidate(rdRecord)) {
```

Comment:
If there is an RD that does not have a future Opp but it has a past Opp, a new Opp will be created (if all conditions align for a new record) and the past Opp will be verified to ensure currency is matching, so there can be a scenario where both Opp insert and update will be done for the same RD.